# HG changeset patch
# User Rik <rik@octave.org>
# Date 1522975155 25200
#      Thu Apr 05 17:39:15 2018 -0700
# Branch stable
# Node ID 0cd7076a97c8258f3f7cc57988225bcd90353994
# Parent  2bc071fc7f6bf7a29a8301709a70908035a148dd
pkg.m: Don't leave behind temporary directories if error occurs during install (bug #53567).

* pkg.m: Change unwind_protect_cleanup to use rmdir with recursive option to
delete both temporary files and the temporary directory they were in.

diff --git a/scripts/pkg/pkg.m b/scripts/pkg/pkg.m
--- a/scripts/pkg/pkg.m
+++ b/scripts/pkg/pkg.m
@@ -432,17 +432,19 @@ function [local_packages, global_package
 
             endfor
           endif
         endif
         install (files, deps, prefix, archprefix, verbose, local_list,
                  global_list, global_install);
 
       unwind_protect_cleanup
-        cellfun ("unlink", local_files);
+        if (exist (tmp_dir, "file"))
+          rmdir (tmp_dir, "s");
+        endif
       end_unwind_protect
 
     case "uninstall"
       if (isempty (files))
         error ("pkg: uninstall action requires at least one package name");
       endif
       uninstall (files, deps, verbose, local_list, global_list, global_install);
 
